simGanESS/source/geometries/GaP.cc

```diff
@@ -332,49 +449,173 @@ void GaP::BuildTPC(G4Material* gas, G4Material* mesh_mat, G4LogicalVolume* logic
+    G4VSolid *solidSub_pmtHolder0  = new G4SubtractionSolid("PMTHolder", solid_pmtHolder, solid_pmt,     0,G4ThreeVector(-15.573*mm, -32.871*mm, 0.));
+    G4VSolid *solidSub_pmtHolder1  = new G4SubtractionSolid("PMTHolder", solidSub_pmtHolder0, solid_pmt, 0,G4ThreeVector(20.68*mm,   -29.922*mm, 0.));
+    G4VSolid *solidSub_pmtHolder2  = new G4SubtractionSolid("PMTHolder", solidSub_pmtHolder1, solid_pmt, 0,G4ThreeVector(-36.253*mm, -2.949*mm,  0.));
+    G4VSolid *solidSub_pmtHolder3  = new G4SubtractionSolid("PMTHolder", solidSub_pmtHolder2, solid_pmt, 0,G4ThreeVector(0.,          0.     ,   0.));
+    G4VSolid *solidSub_pmtHolder4  = new G4SubtractionSolid("PMTHolder", solidSub_pmtHolder3, solid_pmt, 0,G4ThreeVector(36.253*mm,   2.949*mm,  0.));
+    G4VSolid *solidSub_pmtHolder5  = new G4SubtractionSolid("PMTHolder", solidSub_pmtHolder4, solid_pmt, 0,G4ThreeVector(-20.68*mm,   29.922*mm, 0.));
+    G4VSolid *solidSub_pmtHolder6  = new G4SubtractionSolid("PMTHolder", solidSub_pmtHolder5, solid_pmt, 0,G4ThreeVector(15.573*mm,   32.871*mm, 0.));
+
+    G4LogicalVolume *logic_pmtHolder = new G4LogicalVolume(solidSub_pmtHolder6, steel, "PMTHolder");
+
+    G4double pmtHolder_z = enclosurevac_pmt_length_/2 - pmtHolder_length_/2;
+    new G4PVPlacement(0, G4ThreeVector(0., 0, pmtHolder_z), logic_pmtHolder, "PMTHolder", logic_enclosurevac_pmt, false, 0, true);
+
+    // Steel plate enclosing the pmt tube
+    G4Tubs *solid_plate1_pmt = new G4Tubs("PMTplate1", 0, plate_pmt_rad_+plate_pmt_thickn_, plate_pmt_length_/2, 0, 360*deg);
+    G4VSolid *solidSub_plate1_pmt0  = new G4SubtractionSolid("PMTplate1", solid_plate1_pmt, solid_pmt,     0,G4ThreeVector(-15.573*mm,  -32.871*mm, 0.));
+    G4VSolid *solidSub_plate1_pmt1  = new G4SubtractionSolid("PMTplate1", solidSub_plate1_pmt0, solid_pmt, 0,G4ThreeVector(20.68*mm,    -29.922*mm, 0.));
+    G4VSolid *solidSub_plate1_pmt2  = new G4SubtractionSolid("PMTplate1", solidSub_plate1_pmt1, solid_pmt, 0,G4ThreeVector(-36.253*mm,  -2.949*mm,  0.));
+    G4VSolid *solidSub_plate1_pmt3  = new G4SubtractionSolid("PMTplate1", solidSub_plate1_pmt2, solid_pmt, 0,G4ThreeVector(0.,           0.,        0.));
+    G4VSolid *solidSub_plate1_pmt4  = new G4SubtractionSolid("PMTplate1", solidSub_plate1_pmt3, solid_pmt, 0,G4ThreeVector(36.253*mm,    2.949*mm,  0.));
+    G4VSolid *solidSub_plate1_pmt5  = new G4SubtractionSolid("PMTplate1", solidSub_plate1_pmt4, solid_pmt, 0,G4ThreeVector(-20.68*mm,    29.922*mm, 0.));
+    G4VSolid *solidSub_plate1_pmt6  = new G4SubtractionSolid("PMTplate1", solidSub_plate1_pmt5, solid_pmt, 0,G4ThreeVector(15.573*mm,    32.871*mm, 0.));
+
+    G4LogicalVolume *logic_plate1_pmt = new G4LogicalVolume(solidSub_plate1_pmt6, steel, "PMTplateBottom");
+
+    G4double plate1_pmt_z = enclosure_pmt_z - enclosure_pmt_length_/2 - plate_pmt_length_/2;
+    new G4PVPlacement(0, G4ThreeVector(0., 0., -plate1_pmt_z), logic_plate1_pmt, "PMTplateBottom", logic_vessel, false, 0, true);
```

Comment:
In general all these lines (from the declaration of the PMTs up to here) seem non optimal. I think this should be more clean and would make more sense in a for loop (1 loop per PMT). See if you can modify the code to do that, otherwise it's too crowded.

Reply:
Well, maybe up to a bit later, you'll see once you do it.